Sketch.java

```diff
@@ -1,36 +1,137 @@
 import processing.core.PApplet;
 
 public class Sketch extends PApplet {
-	
-	
-  /**
-   * Called once at the beginning of execution, put your size all in this method
-   */
+
   public void settings() {
-	// put your size call here
-    size(400, 400);
+    // put your size call here
+    size(300, 300);
   }
 
-  /** 
-   * Called once at the beginning of execution.  Add initial set up
-   * values here i.e background, stroke, fill etc.
-   */
+  float[] snowflakeY = new float[35];
+  boolean upPressed = false;
+  boolean downPressed = false;
+  boolean leftPressed = false;
+  boolean rightPressed = false;
+
+  float playerX = 200;
+  float playerY = 200; 
+
+  float[] clicked = new float[5];
+
+  float clickedX;
+  float clickedY;
+  float newclick;
+
+  int count = 0;
+
   public void setup() {
-    background(210, 255, 173);
+    for (int i = 0; i < snowflakeY.length; i++) {
+      snowflakeY[i] = random(height);
+    }
   }
 
-  /**
-   * Called repeatedly, anything drawn to the screen goes here
-   */
   public void draw() {
-	  
-	// sample code, delete this stuff
-    stroke(128);
-    line(150, 25, 270, 350);  
 
-    stroke(255);
-    line(50, 125, 70, 50);  
+    background(50);
+
+    float z = 2;
+
+    for (int i = 0; i < snowflakeY.length; i++) {
+
+      float snowflakeX = width * i / snowflakeY.length;
+      
+      fill(225);
+      ellipse(snowflakeX, snowflakeY[i], 10, 10);
+      
+      if (snowflakeX - clickedX < 2 && snowflakeY[i] - clickedY < 5) {
+        snowflakeY[i] = newclick;
+        fill(50);
+        ellipse(clickedX, newclick, 15, 15);
+      }
+      
+      snowflakeY[i] += z;
+      newclick += z;
+
+      if (snowflakeY[i] > height) {
+        snowflakeY[i] = 0;
+      }
+
+      if (keyPressed) {
+        if (keyCode == DOWN) {
+          z = 3;
+        }
+        if (keyCode == UP) {
+          z = 1;
+        }
+      }
+
+      if (upPressed) {
+        playerY--;
+      }
+      if (downPressed) {
+        playerY++;
+      }
+      if (leftPressed) {
+        playerX--;
+      }
+      if (rightPressed) {
+        playerX++;
+      }
+
+      fill(0, 0, 255);
+      ellipse(playerX, playerY, 20, 20);
+
+      if (dist(playerX, playerY, snowflakeX, snowflakeY[i]) < 3 && dist(playerX, playerY, snowflakeX, snowflakeY[i]) > 2) {
+        count++;
+      }
```

Comment:
It looks like collision isn't functioning fully as expected.  Colliding with a snowball doesn't always result in a loss of life when it should.